kie-api/src/main/java/org/kie/api/builder/ReleaseIdComparator.java

```diff
@@ -144,12 +139,12 @@ public int compareTo(Item item) {
                         return 1; // 1.1 > 1-1
 
```

Comment:
you are not checking both items are the same type and how it behaves when the compare mixing ... what is first an Integer + null or it should be sort out as null + integer ? and so forth
@gmunozfe plz check this sort of stuff with francisco

Reply:
the logic is similar to the previous one, if the object being compared to is null and the Integer Type is "null", they are equal. If Integer Type is not null, the it is greater, like in the case the object being compared to is String or List, them Integer is considered greater.
I will add comment to the description explaining better what was changed